frontend/src/app/libs/admin-role-management/components/users-role-management-panel/users-role-management-panel.component.ts

```diff
@@ -8,57 +8,150 @@ import { AuthFacadeService } from 'src/app/libs/auth/store/auth/auth.facade';
+
 	constructor(
-		private userSearchService: UserSearchService,
 		private usersFacadeService: UsersFacadeService,
-		private authFacadeService: AuthFacadeService
+		private authFacadeService: AuthFacadeService,
 	) {}
 
-	public get updateInProgress(): boolean {
-		return this.userUpdate.status === EntityStatus.Pending;
-	}
+	public ngOnInit(): void {
 
-	public get foundUsers(): User[] {
-		return this.userSearchService.userSearch(this.users, this.searchText);
-	}
+		this.usersFacadeService.users$.pipe(takeUntil(this.destroy$)).forEach((users: EntityWrapper<User[]>) => {
 
-	public get foundUsersEmpty(): boolean {
-		return this.foundUsers && this.foundUsers.length === 0;
-	}
+			this.usersRequestStatus = users.status;
 
-	public ngOnInit(): void {
-		this.usersFacadeService.users$.pipe(takeUntil(this.destroy$)).forEach((users: EntityWrapper<User[]>) => {
-			this.users = users.value;
+			const temporaryUsers: User[] = [];
+
+			if (Boolean(users.value?.length) && this.usersRequestStatus !== EntityStatus.Pending) {
+				for (const user of users.value) {
+					let userForTable: User = {...user};
+
+					if (Boolean(this.selectedUsers?.length)) {
+						userForTable = this.getOriginalSelectedUserObjects(userForTable);
+					}
+
+					userForTable.isEventManager =
+						userForTable.claims.includes(UserClaim.EventManager) ?
+						true :
+						false;
+
+					temporaryUsers.push(userForTable);
+				}
+
+				this.users = temporaryUsers;
+
+				if (Boolean(this.tableComponent) && this.isSelectedUsersRemoving) {
+					this.isSelectedUsersRemoving = false;
+					this.selectedUsers = [];
+					this.tableComponent.clearSelectedRows();
+				}
+			}
 		});
 
 		this.authFacadeService.user$.pipe(takeUntil(this.destroy$)).subscribe((user: EntityWrapper<User>) => {
 			this.userUpdate = user;
 		});
 	}
 
-	public toggleEventManagerClaim(isEventManager: boolean, user: User): void {
-		const toggledUser: User = isEventManager ?
-						{...user, claims: [...user.claims, UserClaim.EventManager]} :
-						{...user, claims: user.claims.
-							filter((claim: UserClaim) => claim !== UserClaim.EventManager)};
-		this.usersFacadeService.toggleEventManagerClaim(toggledUser);
+	// we need original selected objects for comparison in table component to render this selections
+	public getOriginalSelectedUserObjects(user: User): User {
+		for (const selectedUser of this.selectedUsers) {
+			if (user.id === selectedUser.id) {
+				user = selectedUser;
+			}
+		}
+		return user;
 	}
 
 	public ngOnDestroy(): void {
 		this.destroy$.next(true);
 		this.destroy$.complete();
 	}
+
+	public get updateInProgress(): boolean {
+		return this.userUpdate.status === EntityStatus.Pending;
+	}
+
+	public toggleEventManagerClaim(event: RowToggleOutput<User>): void {
+		const user: User = event.row;
+		const toggledUser: User = event.state ?
+						{...user, claims: [...user.claims.
+							filter((claim: UserClaim) => claim !== UserClaim.EventManager), UserClaim.EventManager]} :
```

Comment:
Looks like we able to encapsulate filter logic in one place and avoid duplications.